### PR TITLE
Add basic webpack for JS bundle

### PR DESCRIPTION
**Pull request by mdo**
This is only a first stab at adding webpack bundling for our JS here. I know we could load up more through this, like our styles, but I'm unsure of the immediate benefit to that.

Also want to use this as a brief opportunity to figure out if webpack is the right choice here, or if Rollup or something else is. I'm still familiarizing myself with some of the tooling and concerns around requiring the transpilers since the regular imports we have here today seem to work.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -33,6 +33,6 @@ <h5 class="modal-title" id="exampleModalLabel">Success!</h5>
     </div>
 
     <script src="node_modules/jquery/dist/jquery.slim.min.js"></script>
```
Comment (Johann-S) [inline index.html:35]: there is no need to import jQuery here, it's already imported in your bundle because Bootstrap import jQuery

**Comment by Johann-S** [comment]
Webpack is a good choice, most of the time Rollup is used to build libs and Webpack is used a lot

**Comment by HansUXdev** [comment]
Does this really need a webpack config file?
We could use use the webpack cli.


```json
  "config": {
    "src": "./assets/js/src",
    "dest": "./assets/js/dist",
    "presets":"@babel/preset-env",
    "plugins":"@babel/plugin-proposal-optional-chaining,@babel/plugin-proposal-function-bind",
    "mode":"production",
    "targets":"browserslist:modern",
    "ignore ": "",
  },
  "scripts": {
    "babel": "babel $npm_package_config_src --out-dir $npm_package_config_dest --presets=$npm_package_config_presets --plugins $npm_package_config_plugins",
    "js": "npx webpack serve --entry $npm_package_config_src/starter.js  ---output-path $npm_package_config_dest --mode development",
    "js:prod": "npx webpack --entry $npm_package_config_src/starter.js  ---output-path $npm_package_config_dest  --env prod --env min",
  },
```
Something like this could also help solve #21 by using the built in dev server which has live reloading. Also the `babel` task is meant to give users more controller over their JS targets and runs before webpack.

**Comment by yurj** [comment]
At least with webpack we could deploy the starter compiled files somewhere (index + css + js) without copying all the node_modules directory or loading resources manually. However, compiling can take time when there are a lot of files, shouldn't be the case for npm-starter.

**Pull request closed without merging**